cEP-0019.md

```diff
@@ -0,0 +1,454 @@
+their reviews.
+
+N1 = total number of THUMBS_DOWN 👎 a person gets for all reviews he did.
+
+N2 = total number of THUMBS_DOWN 👎 a person gives to other people for
+their reviews.
+
+Note P2 and N2 are used to encourage people to get involved in this
+reactions feedback system.
+
+Then final score S would be:
+
+S = k * (P1 - N1) + c1 * P2 + c2 \* N2
+
+c1 = 0.2, c2 = 0.4, are coefficients. c2 is larger than c1 because people
+are usually reluctant to give negative reactions.
+
+k is the impact factor. The higher score a person has, the more impacts he has,
+thus his meta-reviews are more valuable. k is based on the scores in the
+previous iteration and is normalized. For example, in a previous iteration,
+Alice got 2 marks, Bob got 0.8 marks and Charlie got 10 marks. The calculation
+procedure would be as follows:
+
+```python
+>>> c = [2, 0.8, 10]
+>>> max_score = max(c)
+>>> min_score = min(c)
+>>> result = [(i - min_score) / (max_score - min_score) for i in c]
+>>> print(result)
+[0.13043478260869565, 0.0, 1.0]
+>>> result_adjust = [i + 0.1 for i in result] # add bias
+>>> print(result_adjust)
+[0.23043478260869565, 0.1, 1.1]
+>>> result_rounded = [round(i, 3) for i in result_adjust]
+>>> print(result_rounded)
+[0.23, 0.1, 1.1]
+```
+
+Therefore, the k factor for Alice, Bob and Charlie in the next iteration would
+be: 0.23, 0.1 and 1.1, respectively.
+
+If a person has zeros for all factors, then he is regarded as an inactive
+reviewer and would not be displayed. If a person has negative score, he will
+still be displayed, but the meta-reviews he gives to other people will not
+count.
+
+#### Meta-reviews in Need
+
+##### Definition
+
+When is a review needing meta-review? There are two perspectives:
+
+1. Meta-review by developers, i.e. all people within the organization. The
+   'developer' concept here is different from that used in coala. Instead,
+   anyone with write access (including maintainers) to the repo should be
+   regarded as developers. For developers, review comments before merge/close
+   need meta-reviews. Those review comments are sorted according to update
+   time. In particular, one will always see the review comment that is under
+   his PR first.
+
+2. Meta-review by maintainers. For maintainers, review comments after
+   merge/close also need meta-reviews. Those meta-reviews will reflect the
+   accuracy of the review comments.
+
+##### Display
+
+Review comments that need meta-review will be displayed on a separate column
+on gh-board. There are three approaches:
+
+1. The worst case is that the tracking stores in the json which
+   PR do not have meta-reviews, and gh-board shows only a list of those PRs
+   (and the user must then find the reviews in GitHub PR web app, and do
+   meta-review there).
+
+2. An improvement is to store the identifier of the PR comments which
+   do not have meta-reviews and gh-board links to the PR review comment
+   in GitHub PR web app.
```

Comment:
doing it in GitHub has the benefit of letting the user have their usual tools, e.g. user-scripts or https://github.com/sindresorhus/refined-github